src/vm/comsynchronizable.cpp

```diff
@@ -473,7 +477,7 @@ void ThreadNative::StartInner(ThreadBaseObject* pThisUNSAFE)
 
         BOOL success = pNewThread->CreateNewThread(
                                         pNewThread->RequestedThreadStackSize() /* 0 stackSize override*/,
-                                        KickOffThread, share);
+                                        KickOffThread, share, gc.name == NULL ? NULL : gc.name->GetBuffer());
 
```

Comment:
`CreateNewThread` has `GC_TRIGGERS` contract. Passing unpinned string to it is potential GC hole.

Reply:
My understanding is that this is MODE_COOPERATIVE, which means while GCs can happen, they can only happen at places where code explicitly allows them (that GC mode switch points) and thus not during the call to SetThreadName.  

But I do agree that it is fragile (if the mode gets set to PREEMPTIVE we would have a GC hole), and that in general calling external (OS) functions in MODE_COOPERATIVE is not a good thing.  

What is the guidance?    I did not see a easy way of pinning.  I can copy it out to a SString.  Is that what you suggest?

Reply:
Yes, we typically copy the string out in cases like these.